pkg/extension/sumologicextension/extension.go

```diff
@@ -592,6 +618,135 @@ func (se *SumologicExtension) sendHeartbeatWithHTTPClient(ctx context.Context, h
 	return nil
 }
 
+func getHostIpAddress() (string, error) {
+	c, err := net.Dial("udp", "8.8.8.8:53")
+	if err != nil {
+		return "", err
+	}
+
+	defer c.Close()
+	a := c.LocalAddr().(*net.UDPAddr)
+	h, _, err := net.SplitHostPort(a.String())
+	if err != nil {
+		return "", err
+	}
+
+	return h, nil
+}
+
+func (se *SumologicExtension) updateMetadataWithHTTPClient(ctx context.Context, httpClient *http.Client) error {
+	u, err := url.Parse(se.BaseUrl() + metadataUrl)
+	if err != nil {
+		return fmt.Errorf("unable to parse metadata URL %w", err)
+	}
+
+	info, err := host.Info()
+	if err != nil {
+		return err
+	}
+
+	ip, err := getHostIpAddress()
+	if err != nil {
+		return err
+	}
+
+	td := se.conf.CollectorFields
+	if td == nil {
+		td = map[string]interface{}{}
+	}
+
+	var buff bytes.Buffer
+	if err = json.NewEncoder(&buff).Encode(api.OpenMetadataRequestPayload{
+		HostDetails: api.OpenMetadataHostDetails{
+			Name:        info.Hostname,
+			OsName:      info.OS,
+			OsVersion:   info.PlatformVersion,
+			Environment: se.conf.CollectorEnvironment,
+		},
+		CollectorDetails: api.OpenMetadataCollectorDetails{
+			RunningVersion: "1.0.0",
+		},
+		NetworkDetails: api.OpenMetadataNetworkDetails{
+			HostIpAddress: ip,
+		},
+		TagDetails: td,
+	}); err != nil {
+		return err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), &buff)
+	if err != nil {
+		return fmt.Errorf("unable to create HTTP request %w", err)
+	}
+
+	addJSONHeaders(req)
+
+	se.logger.Info("Updating collector metadata",
+		zap.String("URL", u.String()),
+		zap.String("body", buff.String()))
```

Comment:
If we're going to log this, we should make sure that it's human-readable. Setting an indentation level on the encoder should suffice.

Reply:
Zap String() doesn't pretty print it as expected, for example: `{\n\t\t\"runningVersion\":`

Reply:
This logging is really for our own testing, it should move to debug or be removed when we remove the feature gate.